packages/server/src/core/middleware.ts

```diff
@@ -137,7 +137,7 @@ export type MiddlewareFunction<
     meta: TParams['_meta'] | undefined;
     next: {
       (): Promise<MiddlewareResult<TParams>>;
-      <$Context>(opts: { ctx: $Context }): Promise<
+      <$Context>(opts: { ctx?: $Context; input?: unknown }): Promise<
```

Comment:
I am about 99.9% I tried this but it had some edge-case.... but I can't remember now and it evidently passes tests 🤔 

Can you make sure it works if you downgrade to TS 4.7?

Reply:
for sure, i shall do that :)

Reply:
@KATT so I downgraded typescript to 4.7.4 and i found the current error in the `fastifyRequestHandler.ts` file:

![image](https://user-images.githubusercontent.com/20271968/222851471-3727780a-b399-4948-a721-ae44c5c44513.png)

this is however not connected to the following change of this PR as that I can confirm the error shows up on the base `next` branch. I was able to fix the error by doing the following:

![image](https://user-images.githubusercontent.com/20271968/222851657-f402cd42-5193-4580-9e43-c79404a1783a.png)

So in conclusion it doesn't seem like my change does produce any type errors on Typescript version `4.7.4` and in the process I discovered an unrelated error that's easily solveable.